### DISCUSSION TITLE
LAB 02.02 - Numpy

### DISCUSSION THREAD

**Discussion by luthors** [rramosp/ai4eng.v1#35] [02 - PYTHON]
Hola a todos...  les agradezco sus opiniones
Tengo este código que funciona muy bien a la hora de hacer los cálculos que pide el ejercicio Task 1: Cauchy Matrix, pero no me captura el except,, aun si lo dejo solo sin el  ValueError.    Alguien me podría ayudar a encontrar mi error
def cauchy(x, y):
    try:
        r=np.zeros((len(x), len(y)))+(1/((x[:].reshape(-1,1))-(y[:].reshape(1,-1))))
    except ValueError:
        return False
    return  r

**Comment by rramosp** [comment]
hola luthor, no tienes que capturar un ValueError. Es al revés, tienes que detectar la condición de división por cero (con un if) y, si la detectas, entonces tienes que **lanzar** un ValueError con `raise`

**Reply by luthors** [reply]
Muchas Gracia profesor por su pronta respuesta
